lib/internal/webstreams/readablestream.js

```diff
@@ -638,23 +639,13 @@ ObjectDefineProperties(ReadableStream, {
   from: kEnumerableProperty,
 });
 
-function InternalTransferredReadableStream() {
-  markTransferMode(this, false, true);
-  this[kType] = 'ReadableStream';
-  this[kState] = createReadableStreamState();
-
-  this[kIsClosedPromise] = createDeferredPromise();
-}
-
-ObjectSetPrototypeOf(InternalTransferredReadableStream.prototype, ReadableStream.prototype);
-ObjectSetPrototypeOf(InternalTransferredReadableStream, ReadableStream);
-
 function TransferredReadableStream() {
-  const stream = new InternalTransferredReadableStream();
-
-  stream.constructor = ReadableStream;
-
-  return stream;
+  return ReflectConstruct(function() {
```

Comment:
ReflectConstruct is extremely slow.

Reply:
I'm aware, is there another way to make this WPT compliant?

Reply:
FWIW regarding speed, this has around a -20% on speed, so not good.

Reply:
Deno solves this by [adding a special overload](https://github.com/denoland/deno/blob/0cb00a6e89d83d4e16e6616f7af8819bd894b0da/ext%2Fweb%2F06_streams.js#L5141-L5144) to their `ReadableStream` constructor, which essentially skips the regular steps. Maybe we should do something similar?